src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketSCMSource.java

```diff
@@ -823,9 +827,19 @@ private void retrieveTags(final BitbucketSCMSourceRequest request) throws IOExce
         request.listener().getLogger().format("%n  %d tags were processed%n", count);
     }
 
+    private void trackCredentialsUsage() {
+        final SCMSourceOwner owner = getOwner();
+        if (owner != null) {
+            CredentialsProvider.track(owner, credentials());
+        }
+    }
+
     @Override
     protected SCMRevision retrieve(SCMHead head, TaskListener listener) throws IOException, InterruptedException {
         final BitbucketApi bitbucket = buildBitbucketClient();
+
+        trackCredentialsUsage();
```

Comment:
It could, but my intention was to make this credentials tracking very explicit and not hide it as a side effect of a method where you wouldn't necessary expect it

I can move it there though if you think it is a better place
one advantage woudl be that itt would also track the usage as part of the `BitbucketGitScmBuilder` alhtough the `authenticator()` method is called there mutliple times (as far as I was able to tell from a quickl glance at the code once for the primary and each mirror for PRs, branches, tags respectively) - but to be honest I'm not really sure when exactly this code is executed yet - didn't dig trough that yet

Reply:
When looking through the code again today I actually figured out that I was still missing a couple of usages when the credentials have been used in the `BtibucketSCMNavigator` - after checking I figured out that the logic of credential tracking could actullly be moved to `BitbucketCredentials#lookupCredentials` where it captures everything and we already have all the required information available. I'll move the tracking there.
Personally I'd also be in favor of renaming the method from `lookupCredentials` to `lookupCredentialsAndTrackUsage` or something simlar as well to make it very clear that this is happening here, but I'll await your feedback on that - for now I'll just add this information to the javadoc